Orchestrator.py

```diff
@@ -495,3 +565,159 @@ def _isCornerFrontier(self, cell):
                         break
 
         return returnVal
+
+    # === Exploration
+
+    def _computeTargetCell(self, dotBotId):
+        dotBot         = self.dotBotsView[dotBotId]
+        targetFrontier =  None
+
+        if self.cellsFrontier:
+            # find closest frontiers to initial position
+            cellsAndDistancesToStart = [((cx, cy), u.distance((self.initX, self.initY), (cx, cy))) for (cx, cy) in
+                                        self.cellsFrontier if (cx, cy) not in self.assignedFrontiers]
+
+            if cellsAndDistancesToStart:
+                cellsAndDistancesToStart = sorted(cellsAndDistancesToStart, key=lambda e: e[1])
+                closestFrontiersToStart  = [cell for (cell, distance) in cellsAndDistancesToStart if
+                                           distance == cellsAndDistancesToStart[0][1]]
+
+                # find closest frontier to robot
+                cellsAndDistancesToDotBot = [((cx, cy), u.distance((dotBot['x'], dotBot['y']), (cx, cy))) for (cx, cy) in
+                                             closestFrontiersToStart]
+                cellsAndDistancesToDotBot = sorted(cellsAndDistancesToDotBot, key=lambda e: e[1])
+                closestFrontiersToDotBot  = [cell for (cell, distance) in cellsAndDistancesToDotBot if
+                                            distance == cellsAndDistancesToDotBot[0][1]]
+                targetFrontier            = random.choice(closestFrontiersToDotBot)
+                self.assignedFrontiers   += [targetFrontier]
+
+        return targetFrontier
+
+    # === Navigation
+
+    def _computePath(self, startCell, targetCell, excludeDiagonalCells=False):
```

Comment:
I don't have time to read in detail. Add a comment which shows an example input and output and 1 sentence which describes how this function works